File: Python/some_functions.py
def arg_sort(a: list, reverse=False) -> list[int]:
    """Takes a list and retuns a list contains argmunt sort

    :param a: list that need to be sorted
    :type a: list
    :param reverse: if True the result will be in descending order, defaults to False
    :type reverse: bool, optional
    :return: a list of sorted arguments
    :rtype: list[int]
    """
    return sorted(range(len(a)), key=a.__getitem__, reverse=reverse)

File: Cpp/some_functions.cpp
#include <vector>
#include <algorithm>

std::vector<int> arg_sort(std::vector<int> a)
{
    std::vector<int> a_index;
    for (int i = 0; i < a.size(); i++)
    {
        a_index.push_back(i);
    }

    std::sort(a_index.begin(), a_index.end(),
              [&a](size_t i1, size_t i2)
              { return a[i1] < a[i2]; });

    return a_index;
}